src/NuGet.Clients/NuGet.SolutionRestoreManager/VSNominationUtilities.cs

```diff
@@ -712,5 +712,44 @@ private static string[] HandleClear(string[] input)
 
             return input;
         }
+
+        internal static NuGetVersion? GetSdkAnalysisLevel(IReadOnlyList<IVsTargetFrameworkInfo4> targetFrameworks)
+        {
+            string skdAnalysisLevelString = GetSingleNonEvaluatedPropertyOrNull(targetFrameworks, nameof(ProjectBuildProperties.SdkAnalysisLevel), v => v) ?? string.Empty;
+            NuGetVersion? sdkAnalysisLevel = null;
+
+            if (!string.IsNullOrEmpty(skdAnalysisLevelString))
+            {
+                try
+                {
+                    sdkAnalysisLevel = new NuGetVersion(skdAnalysisLevelString);
+                }
+                catch (ArgumentException ex)
+                {
+                    throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Strings.Invalid_AttributeValue, ProjectBuildProperties.SdkAnalysisLevel, skdAnalysisLevelString, "9.0.100"), ex);
+                }
+            }
+
+            return sdkAnalysisLevel;
+        }
+
+        internal static bool GetUsingMicrosoftNETSdk(IReadOnlyList<IVsTargetFrameworkInfo4> targetFrameworks)
+        {
+            string? usingNetSdk = GetSingleNonEvaluatedPropertyOrNull(targetFrameworks, nameof(ProjectBuildProperties.UsingMicrosoftNETSdk), v => v);
+
+            if (!string.IsNullOrEmpty(usingNetSdk))
+            {
+                if (bool.TryParse(usingNetSdk, out var result))
+                {
+                    return result;
+                }
+                else
+                {
+                    throw new ArgumentException(string.Format(CultureInfo.CurrentCulture, Strings.Invalid_AttributeValue, ProjectBuildProperties.UsingMicrosoftNETSdk, usingNetSdk, "false"));
+                }
+            }
+
+            return true;
```

Comment:
It feels unintuitive to have a true as the default value of something such as UsingNetSDK. 

I think this isn't related to the roundtripping we do for the package spec, so I think false is the right thing here.